assets/src/js/Components/ArchivePicker/ArchivePicker.js

```diff
@@ -309,6 +314,18 @@ export default function ArchivePicker({view = ADMIN_VIEW}) {
         await wp.data.dispatch('core/block-editor').updateBlock(currentBlock.clientId, updatedAttributes);
       }
 
+      if (currentBlock.name === 'core/media-text') {
```

Comment:
I noticed both if-else are basically doing the same. Then I think we can do this 👇 instead
```suggestion
      if (currentBlock.name === 'core/media-text' || || currentBlock.name === 'planet4-blocks/carousel-header') {
        const uploadedImage = await getImageDetails(id);
        const updatedAttributes = updateCarouselBlockAttributes(uploadedImage, currentBlock);
        await wp.data.dispatch('core/block-editor').updateBlock(currentBlock.clientId, updatedAttributes);
      }
      }
```

What do you think?

Reply:
Well they are not doing the same thing, if you look they both call different functions that updated the attributes differently.

Reply:
It seems that at least two of these lines of code are repeated so maybe there's a way to remove that duplication?
```
const uploadedImage = await getImageDetails(id);
```
and
```
await wp.data.dispatch('core/block-editor').updateBlock(currentBlock.clientId, updatedAttributes);
```

Reply:
It seems it would be possible to write something like:
```
const updatedAttributes = currentBlock.name === 'core/media-text' ? updateMediaAndTextAttributes(uploadedImage, currentBlock) : updateCarouselBlockAttributes(uploadedImage, currentBlock);
```